### PR TITLE
Create Select component for Formik consumption of react-select

### PR DESCRIPTION
**Pull request by kylemh**
# Description of changes

Creates a Formik-ready Select component that leverages `react-select`.

We should now have all the pieces we need to finish any/all forms on the website.

#### Drive-By Changes
- Removed Form.css as it was unused
- Adjust specific test in Input's test file

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit d9410ee18c2d09554525aa576873586afcc63625

https://deploy-preview-469--operation-code-storybook.netlify.com

File: components/Form/Select/Select.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        <Label for={name} isHidden={isLabelHidden}>
+          {label}
+        </Label>
+
+        <div>
+          <ThemedReactSelect
+            hasErrors={hasErrors}
+            isTouched={touched[name]}
+            id={id}
+            isDisabled={disabled}
```
Comment (kylemh) [inline components/Form/Select/Select.js]: Rotate `disabled` with `isDisabled`

File: components/Form/Select/Select.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  static propTypes = {
+    disabled: bool,
+    field: shape({
+      name: string.isRequired,
+      value: oneOfType([
+        string,
+        arrayOf(shape({ label: string.isRequired, value: string.isRequired })),
+      ]),
+    }).isRequired,
+    form: shape({
```
[the remark this answers is not in the document]
Comment (kylemh) [reply components/Form/Select/Select.js]:
```javascript
<Field
  type="email"
  name="email"
  label="Email*"
  component={Input}
  disabled={isSubmitting}
  autoComplete="username email"
/>
```

It's easier to understand the "why" when you see an implementation.

Basically, `FIeld` has required props which are passed along as an object (`field`). Other props get passed thru, even when they're not related to the component.

tl;dr kind of necessary rather than by choice. we could put the component in a render prop to pass the specific props, but it's nice having a catch-all since the potential props are large. I'm just tryna avoid documenting all of Formik's API.

File: components/Form/Select/__stories__/Select.stories.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            errors: { [selectName]: '' },
+            setFieldTouched: action('setFieldTouched'),
+            setFieldValue: action('setFieldValue'),
+          })}
+          id={text('id', '')}
+          isLabelHidden={boolean('isLabelHidden', false)}
+          isMulti={isMulti}
+          label={text('label', 'Some Select:')}
+          options={options}
+          placeholder={text('placeholder', '')}
```
[the remark this answers is not in the document]
Comment (kylemh) [reply components/Form/Select/__stories__/Select.stories.js]: Fixed!

File: components/Form/Select/__stories__/Select.stories.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          field={{ name: selectName }}
+          form={object('form', {
+            touched: { [selectName]: false },
+            errors: { [selectName]: '' },
+            setFieldTouched: action('setFieldTouched'),
+            setFieldValue: action('setFieldValue'),
+          })}
+          id={text('id', '')}
+          isLabelHidden={boolean('isLabelHidden', false)}
+          isMulti={isMulti}
```
[the remark this answers is not in the document]
Comment (kylemh) [reply components/Form/Select/__stories__/Select.stories.js]: Fixed!
